python/dgl/nn/pytorch/hetero.py

```diff
@@ -250,3 +250,137 @@ def get_aggregate_fn(agg):
         return _stack_agg_func
     else:
         return partial(_agg_func, fn=fn)
+
+class HeteroLinearLayer(nn.Module):
+    """Apply a linear transformation on the node features of a
+    heterogeneous graph.
+
+    Parameters
+    ----------
+    hg : DGLGraph
+        A heterogeneous graph.
+    out_size : int
+        Output feature size.
+    feat_name : str
+        Name of the node feature to transform.
+
+    Examples
+    --------
+
+    >>> import dgl
+    >>> import torch
+    >>> from dgl.nn import HeteroLinearLayer
+    >>> g = dgl.heterograph({
+    ...     ('A', 'r1', 'B'): ([0, 1], [1, 2]),
+    ...     ('B', 'r2', 'B'): ([1], [2])
+    ... })
+    >>> g.nodes['A'].data['h'] = torch.randn(2, 3)
+    >>> g.nodes['B'].data['h'] = torch.randn(3, 4)
+    >>> layer = HeteroLinearLayer(g, 5, 'h')
+    >>> out_feats = layer(g)
+    >>> print(out_feats['A'].shape)
+    torch.Size([2, 5])
+    >>> print(out_feats['B'].shape)
+    torch.Size([3, 5])
+    """
+    def __init__(self, hg, out_size, feat_name):
+        super(HeteroLinearLayer, self).__init__()
+
+        self.feat_name = feat_name
+        self.linears = nn.ModuleDict()
+        for ntype in hg.ntypes:
+            linear = nn.Linear(hg.nodes[ntype].data[feat_name].shape[1], out_size)
+            self.linears[ntype] = linear
+
+    def forward(self, hg):
+        """Forward function
+
+        Parameters
+        ----------
+        hg : DGLGraph
+            The input heterogeneous graph.
+
+        Returns
+        -------
+        dict[str, Tensor]
+            Transformed node features.
+        """
+        feat = dict()
+        for ntype in hg.ntypes:
+            feat[ntype] = self.linears[ntype](hg.nodes[ntype].data[self.feat_name])
+
+        return feat
+
+class HeteroEmbedding(nn.Module):
+    """Create node embeddings for each node type and return a homogeneous
+    graph representation.
+
+    Parameters
+    ----------
+    hg : DGLGraph
+        A heterogeneous graph.
+    embed_size : int
+        Node embedding size.
```

Comment:
Consider follow the naming convention of https://pytorch.org/docs/stable/generated/torch.nn.Embedding.html